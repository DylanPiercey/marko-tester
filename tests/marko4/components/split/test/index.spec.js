const { render, fixtures, createEvent } = require('../../../../..')('../index.marko');

describe('When component is rendered with data', () => {
  let component;

  beforeEach(() => {
    component = render(fixtures.default);
  });

  afterEach(() => {
    component.destroy();
  });

  describe('When user clicks emit button', () => {
    let clickEvent;

    beforeEach(() => {
      clickEvent = createEvent('click');

      jest.spyOn(component, 'emit');
      component.getEl('emit-button').dispatchEvent(clickEvent);
    });

    it('should emit "hello" with event and element', () => {
      expect(component.emit).toBeCalledWith('hello', clickEvent, clickEvent.target);
    });
  });
});
