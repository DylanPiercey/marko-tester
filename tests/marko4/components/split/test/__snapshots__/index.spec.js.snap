// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component with default fixture 1`] = `
Array [
  ,
  <div
    class="mock-component"
  >
    <h1>
      hello
    </h1>
    <button
      type="button"
    />
    <tbody-component
      attribute="{\\"title\\":\\"hello\\",\\"i\\":0}"
    >
      <at_body>
        body is here
      </at_body>
    </tbody-component>
  </div>,
  ,
]
`;
